opm/grid/LookUpData.hh

```diff
@@ -14,51 +14,242 @@
 //===========================================================================
 
 /*
-Copyright 2023 Equinor ASA.
+  Copyright 2023 Equinor ASA.
 
-This file is part of The Open Porous Media project  (OPM).
+  This file is part of The Open Porous Media project  (OPM).
 
-OPM is free software: you can redistribute it and/or modify
-it under the terms of the GNU General Public License as published by
-the Free Software Foundation, either version 3 of the License, or
-(at your option) any later version.
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
 
-OPM is distributed in the hope that it will be useful,
-but WITHOUT ANY WARRANTY; without even the implied warranty of
-MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-GNU General Public License for more details.
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
 
-You should have received a copy of the GNU General Public License
-along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include <dune/grid/common/mcmgmapper.hh>
-#include <opm/grid/cpgrid/Entity.hpp>
+
+#include <type_traits>
+#include <vector>
 
 namespace Dune
 {
-template <typename GridType>
+class CpGrid;
+
+namespace cpgrid
+{
+template<int codim> class Entity;
+}
+
+// LookUpData class - To search data via element index
```

Comment:
I think it would make things easier if we put LookupData into the Opm namespace. Hence close the other one here and open Opm.

Reply:
Done!